UnityProject/Assets/Editor/LevelBuilder/GUI/LevelBuilderWindow.cs

```diff
@@ -102,53 +94,91 @@ public void OnGUI()
-			// added objects through the level builder so we need to force
-			// the change to be saved.
-			EditorSceneManager.MarkAllScenesDirty();
-			EditorSceneManager.SaveOpenScenes ();
+            // We have a bug where Unity doesn't recognise our dynamically 
+            // added objects through the level builder so we need to force
+            // the change to be saved.
+            EditorSceneManager.MarkAllScenesDirty();
+            EditorSceneManager.SaveOpenScenes();
 
-			Scene newScene = SceneHandler.createScene();
+            Scene newScene = SceneHandler.createScene();
 
-			// Unload the scenes so we can create a new one.
-			SceneHandler.unloadScenes();
+            // Unload the scenes so we can create a new one.
+            SceneHandler.unloadScenes();
 
-			// Calling countScenes() inside SceneHandler to find the new 
-			// filename to save our scene with. 
-			int currentNoOfLevels = SceneHandler.countScenes ();
-			currentNoOfLevels++;
+            // Calling countScenes() inside SceneHandler to find the new 
+            // filename to save our scene with. 
+            int currentNoOfLevels = SceneHandler.countScenes();
+            currentNoOfLevels++;
+            SetupCamera();
 
-			// Setting up the camera.
-			GameObject cameraGameObject = new GameObject("Main Camera");
-			cameraGameObject.AddComponent<Camera>();
-			cameraGameObject.tag = "MainCamera";
+            LevelBuilderWindow.SetUpScene();
+            SetupLighting();
 
-			LevelBuilderWindow.SetUpScene ();
+            GameObject level = CreateEmptyLevel();
+            CreateTerrainForLevel(level);
 
-			// Setting up the light.
-			GameObject directionalLight = new GameObject ("Directional Light");
-			Light lightSettings = directionalLight.AddComponent<Light> ();
-			lightSettings.type = LightType.Directional;
+            if (!EditorSceneManager.SaveScene(newScene, "Assets/Scenes/Levels/Level" + currentNoOfLevels + ".unity"))
+            {
+                Debug.Log("The savescene line failed!!!!!!!!!!!!!");
+                // TODO: check for errors here and remove the curly brackets.
+            }
 
-			// The color needs to be normalized to the 0-1 range.
-			lightSettings.color = new Color (1.0f, 0.956f, 0.839f, 1.0f);
-			lightSettings.intensity = 1.0f;
-			lightSettings.shadowBias = 0.05f;
-			lightSettings.shadowNormalBias = 0.4f;
-			lightSettings.shadowNearPlane = 0.2f;
-			directionalLight.transform.position = new Vector3 (0.0f, 3.0f, 0.0f);
-			directionalLight.transform.rotation = Quaternion.Euler(50.0f, -30.0f, 0.0f);
+        } // if
 
-			if (!EditorSceneManager.SaveScene (newScene, "Assets/Scenes/Levels/Level" + currentNoOfLevels + ".unity")) {
-				Debug.Log ("The savescene line failed!!!!!!!!!!!!!");
-				// TODO: check for errors here and remove the curly brackets.
-			}
-
-		} // if
-
-		foreach (IMenu menu in menus) 
+        foreach (IMenu menu in menus) 
 			menu.Display();
```

Comment:
Indentation?

Reply:
the indentation looks fine in the code but I added brackets in to make the if statement clearer